File: 07-Banking-Kata/java/src/main/java/banking/model/Account.java
package banking.model;

import java.util.UUID;

public class Account implements Identifiable {
  
  private final UUID id;
  private Money balance;
  
  private Account(UUID id, Money balance) {
    this.id = id;
    this.balance = balance;
  }

  public Account(Money startingBalance) {
    this(UUID.randomUUID(), startingBalance);
  }
  
  @Override
  public UUID getId() {
    return id;
  }

  public Money getBalance() {
    return balance;
  }

  public Account cloneWithBalance(Money balance) {
    return new Account(this.id, balance);
  }

}

File: 07-Banking-Kata/csharp-dotnetframework/BankingKataAPI/Models/Account.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Katas
{
    public class Account : Identifiable
    {
        public Guid id { get; set; }
        
        [NotMapped]
        public Money balance
        {
            get
            {
                return new Money(DbBalance, "en-US");
            }
            set
            {
                DbBalance = balance.Amount;
            }
        }
        public decimal DbBalance { get; set; }

        private Account(Guid id, Money balance)
		{
			this.id = id;
			DbBalance = balance.Amount;
		}

        public Account(decimal startingBalance)
        {
            this.id = Guid.NewGuid();
            this.DbBalance = startingBalance;
        }

        public Account(Money startingBalance)
		{
            this.id = Guid.NewGuid();
            this.DbBalance = startingBalance.Amount;
		}
        public Account()
        {

        }

        public Guid GetId()
        {
            return id;
        }

		public Account cloneWithBalance(Money balance)
		{
			return new Account(this.id, balance);
		}
    }
}
